app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,14 @@
+class VotesController < ApplicationController
+  def create
+    if params[:voteable_type] == "Question"
+      voteable = Question.find_by(id: params[:question_id])
+    else
+      voteable = Answer.find_by(id: params[:answer_id])
+    end
+    voteable.votes << Vote.new(value: params[:vote_value], user: current_user)
+    if voteable.save && request.xhr?
+      vote_count = voteable.votes_total
+      render json: {vote_count: vote_count}.to_json
+    end
```

Comment:
Always better to provide an else in case of failure.
```
else 
  render text: 'failed', status: 422
end 
```
would trigger any catch / fail on the ajax call